lib/matplotlib/animation.py

```diff
@@ -1156,8 +1160,9 @@ def _init_draw(self):
             self._draw_frame(next(self.new_frame_seq()))
         else:
             self._drawn_artists = self._init_func()
-            for a in self._drawn_artists:
-                a.set_animated(self._blit)
+            if iterable(self._drawn_artists):
```

Comment:
On the other hand, my understanding of the documentation is that the user function are suppossed to return a list of artists.